cmd/common-methods.go

```diff
@@ -32,6 +34,30 @@ import (
 	"github.com/minio/minio-go/v6/pkg/encrypt"
 )
 
+// decode if the key is encoded key and returns the key
+func getDecodedKey(sseKeys string) (key string, err *probe.Error) {
+	keyString := ""
+	for i, sse := range strings.Split(sseKeys, ",") {
+		if i > 0 {
+			keyString = keyString + ","
+		}
+		encryptString := strings.SplitN(sse, "=", 2)
+		pre := encryptString[0]
+		secretValue := encryptString[1]
+		decodedString, e := base64.StdEncoding.DecodeString(secretValue)
+		if e != nil {
+			return "", probe.NewError(errors.New("key should be 32 bytes long or if the key is encoded; then decoded value must be of 32 bytes long"))
+		}
+
+		if len(decodedString) != 32 {
+			keyString = keyString + sse
```

Comment:
shouldn't you be returning an error here because plain text key is not 32 bytes

Reply:
That is taken care in `parseEncryptionKeys` .